### PR TITLE
Add a menu to display groups (based on groupofmember)

### PR DESCRIPTION
**Pull request by olivierguillard**
Alpha version.

This new menu permit to list groups and for each of them, the list of users that belong to it.

best

### CODE REVIEW

File: conf/config.inc.php [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #
 # This program is free software; you can redistribute it and/or
 # modify it under the terms of the GNU General Public License
-# as published by the Free Software Foundation; either version 2
+# as published by the FreeSoftware Foundation; either version 2
```
Comment (coudot) [inline conf/config.inc.php]: This change is not needed

File: conf/config.inc.php [hunk 2]
```diff
@@ -25,14 +25,20 @@
 # LDAP
 $ldap_url = "ldap://localhost";
 $ldap_starttls = false;
-$ldap_binddn = "cn=manager,dc=example,dc=com";
```
Comment (coudot) [inline conf/config.inc.php]: Please don't change default values

File: conf/config.inc.php [hunk 3]
```diff
@@ -25,14 +25,20 @@
...
+$ldap_binddn = "cn=admin,dc=example";
 $ldap_bindpw = "secret";
-$ldap_base = "dc=example,dc=com";
-$ldap_user_base = "ou=users,".$ldap_base;
+$ldap_base = "dc=example";
```
Comment (coudot) [inline conf/config.inc.php]: Please don't change default values

File: conf/config.inc.php [hunk 4]
```diff
@@ -87,19 +93,42 @@
 $gallery_sortby = "lastname";
 $gallery_bootstrap_column_class = "col-xs-6 col-sm-4 col-md-3";
 
-# CSV
-$use_csv = true;
-$csv_filename = "white_pages_export_" . date("Y-m-d") . ".csv";
+###########################################################
```
Comment (coudot) [inline conf/config.inc.php]: You should not remove this

File: conf/config.inc.php [hunk 5]
```diff
@@ -87,19 +93,42 @@
...
+# Full dislpay
+$display_group_items = array('description', 'cn', 'member', 'seealso' );
+$display_group_title = "name";
 
-# vCard
-$use_vcard = true;
-$vcard_file_extension = "vcf";
-$vcard_file_identifier = "identifier";
-$vcard_version = "4.0";
-$vcard_map = array('FN' => 'fullname', 'N' => 'fullname', 'EMAIL' => 'mail', 'CATEGORIES' => 'businesscategory', 'ORG' => 'organization', 'ROLE' => 'employeetype', 'TEL;TYPE=work,voice;VALUE=uri:tel' => 'phone', 'TEL;TYPE=cell,voice;VALUE=uri:tel' => 'mobile', 'UID' => 'identifier');
```
Comment (coudot) [inline conf/config.inc.php]: You should not remove this

File: conf/config.inc.php [hunk 6]
```diff
@@ -87,19 +93,42 @@
...
+# Group
+$use_group = true;
+$group_title = "name";
+$group_sortby = "name";
+$group_bootstrap_column_class = "col-xs-6 col-sm-4 col-md-3";
+######################################################################
 
 # Language
-$lang ="en";
+$lang = "fr";
```
Comment (coudot) [inline conf/config.inc.php]: Don't change default values

File: htdocs/advancedsearch.php [hunk 7]
```diff
@@ -83,26 +83,6 @@
...
-                            $csv_entry[$attribute] = $entry[$attribute][0];
-                        }
-                        $csv_array[] = $csv_entry;
-                }
-                download_send_headers($csv_filename);
-                echo array2csv($csv_array);
-                die();
-            }
-
             if ($nb_entries === 0) {
```
Comment (coudot) [inline htdocs/advancedsearch.php]: Cancel this change

File: htdocs/display.php [hunk 8]
```diff
@@ -35,11 +35,7 @@
         }
         $attributes[] = $attributes_map[$display_title]['attribute'];
 
-        if ($use_vcard and $_GET["vcard"] and $vcard_file_identifier) {
-            $attributes[] = $attributes_map[$vcard_file_identifier]['attribute'];
-	}
-
-	# Search entry
+        # Search entry
```
Comment (coudot) [inline htdocs/display.php]: Cancel this change

File: htdocs/display.php [hunk 9]
```diff
@@ -50,14 +46,6 @@
...
         }
-
-	if ($use_vcard and $_GET["vcard"]) {
-            require_once("../lib/vcard.inc.php");
-            $vcard_file = $entry[0][$attributes_map[$vcard_file_identifier]['attribute']][0].".".$vcard_file_extension;
-            download_vcard_send_headers($vcard_file);
-            echo print_vcard($entry[0], $attributes_map, $vcard_map, $vcard_version);
-            die;
-        }
     }
```
Comment (coudot) [inline htdocs/display.php]: Cancel this change

File: htdocs/index.php [hunk 10]
```diff
@@ -19,7 +19,7 @@
     }
     closedir($handle);
 }
-$lang = detectLanguage($lang, $languages);
+#$lang = detectLanguage($lang, $languages);
```
Comment (coudot) [inline htdocs/index.php]: Cancel this change

File: htdocs/index.php [hunk 11]
```diff
@@ -57,9 +57,23 @@
...
+$smarty->assign('display_group_items',$display_group_items);
+$smarty->assign('display_group_title',$display_group_title);
+$smarty->assign('attributes_group_map',$attributes_group_map);
+$smarty->assign('use_group',$use_group);
+$smarty->assign('group_title',$group_title);
+$smarty->assign('group_bootstrap_column_class',$group_bootstrap_column_class);
+####
 $smarty->assign('date_specifiers',$date_specifiers);
-$smarty->assign('use_csv',$use_csv);
-$smarty->assign('use_vcard',$use_vcard);
```
Comment (coudot) [inline htdocs/index.php]: Cancel this change

File: lang/en.inc.php [hunk 12]
```diff
@@ -1,55 +1,55 @@
 <?php
 
 #==============================================================================
-# English
+# French
 #==============================================================================
```
Comment (coudot) [inline lang/en.inc.php]: Please keep this file for English translation

File: templates/menu.tpl [hunk 13]
```diff
@@ -23,17 +23,26 @@
...
                   <input type="text" class="form-control" placeholder="{$msg_search}" name="search" value="{$search}" />
                   <span class="input-group-btn">
                     <button class="btn btn-default" type="submit">&nbsp;<i class="fa fa-fw fa-search"></i></button>
                   </span>
                 </div>
               </form>
-              {/if}
+	      {/if}
+	      {/if}
             </div>
```
Comment (coudot) [inline templates/menu.tpl]: I don't understand why you do this test in the menu

**Comment by olivierguillard** [comment]
Thanks for the review Clément, I appreciate.

It sounds that I didn't work on the last repo version: I'll check that in the next days.

**Comment by coudot** [comment]
Do what you can, we maybe need to code this feature with more generic code anyway. We also need to apply the export to CSV action to this new kind of objects, this may be interesting.

**Comment by olivierguillard** [comment]
Thanks for the review Clément, I appreciate.

It sounds that I didn't work on the last repo version: I'll check that in the next days.

**Pull request closed without merging**